pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java

```diff
@@ -314,12 +314,24 @@ private static RelDataType inferTypeFromStringLiteral(String operandTypeStr, Rel
     switch (operandTypeStr) {
```

Comment:
Shouldn't the list also contain:
- BOOLEAN
- TIMESTAMP
- FLOAT 
- DOUBLE
?

Reply:
Those are handled in the default branch.